apps/studio/src-commercial/backend/lib/db/clients/mongodb.ts

```diff
@@ -595,29 +604,81 @@ export class MongoDBClient extends BasicDatabaseClient<QueryResult> {
     }
   }
 
-  async executeQuery(queryText: string, _options?: any): Promise<NgQueryResult[]> {
-    const db = this.conn.db(this.db);
+  async getCompletions(cmd: string): Promise<string[]> {
+    log.info('GETTING COMPLETIONS FOR: ', cmd)
+    return (await this.runtime.getCompletions(cmd)).map((c) => c.completion)
+  }
 
-    let result = [];
+  async getShellPrompt(): Promise<string> {
+    return await this.runtime.getShellPrompt()
+  }
 
-    let display = function (data: any) {
-      result.push(this.parseRowQueryResult(data));
+  async executeQuery(queryText: string, _options?: any): Promise<NgQueryResult[]> {
+    let results: NgQueryResult[] = [];
+
+    const listener = {
+      onPrint: (value): void => {
+        value.map((v) => {
+          log.info("PRINT: ", v)
+          results.push({
+            output: v.printable
+          })
+        })
+      }
     }
-    const sandbox = {
-      db,
-      display: display.bind(this)
-    };
 
-    const userScript = `
-      (async () => {
-        ${queryText}
-      })()
-    `
+    this.runtime.setEvaluationListener(listener);
 
-    vm.createContext(sandbox)
-    await vm.runInContext(userScript, sandbox)
+    const ev = await this.runtime.evaluate(queryText);
+    log.info("EVAL: ", JSON.stringify(ev))
 
-    return result;
+    let fields = [];
+    if (ev.type === 'Cursor') {
+      if (ev.printable?.documents?.length > 0) {
+        fields = Object.keys(ev.printable?.documents[0]).map((k) => ({
+          name: k,
+          id: k
+        }));
+      }
+      results.push({
+        rows: ev.printable?.documents,
+        rowCount: ev.printable?.documents?.length,
+        fields,
+        tableName: ev.source?.namespace?.collection ?? 'mycollection',
+        command: queryText
+      })
+    } else if (ev.type === 'Document') {
+      if (ev.printable) {
+        fields = Object.keys(ev.printable).map((k) => ({
+          name: k,
+          id: k
+        }));
+      }
+      results.push({
+        rows: ev.printable ? [ev.printable] : [],
+        rowCount: ev.printable ? 1 : 0,
+        fields,
+        tableName: ev.source?.namespace?.collection ?? 'mycollection',
+        command: queryText
+      })
+    } else if (ev.type === null && ev.printable) {
+      if (ev.printable?.length > 0) {
+        fields = Object.keys(ev.printable[0]).map((k) => ({
+          name: k,
+          id: k
+        }));
+      }
+      results.push({
+        rows: ev.printable,
+        rowCount: ev.printable?.length,
+        fields,
+        tableName: ev.source?.namespace?.collection ?? 'mycollection',
+        command: queryText
+      })
+    }
+    log.info("RESULTS: ", results)
```

Comment:
probably we want this to be dubug.

Any additional tests that we need given the extra code added here?